software/openvisualizer/bin/openVisualizerApp/openVisualizerWeb.py

```diff
@@ -269,62 +286,83 @@ def _topologyMotesUpdate(self):
             if index not in motesTemp:
                 motesTemp[index] = {}
             motesTemp[index][param] = v
-        
+
         for (_,v) in motesTemp.items():
             mh = self.engine.getMoteHandlerById(v['id'])
             mh.setLocation(v['lat'],v['lon'])
-    
+
     def _topologyConnectionsCreate(self):
-        
+
         data = bottle.request.forms
         assert sorted(data.keys())==sorted(['fromMote', 'toMote'])
-        
+
         fromMote = int(data['fromMote'])
         toMote   = int(data['toMote'])
-        
+
         self.engine.propagation.createConnection(fromMote,toMote)
-    
+
     def _topologyConnectionsUpdate(self):
         data = bottle.request.forms
         assert sorted(data.keys())==sorted(['fromMote', 'toMote', 'pdr'])
-        
+
         fromMote = int(data['fromMote'])
         toMote   = int(data['toMote'])
         pdr      = float(data['pdr'])
-        
+
         self.engine.propagation.updateConnection(fromMote,toMote,pdr)
-    
+
     def _topologyConnectionsDelete(self):
-        
+
         data = bottle.request.forms
         assert sorted(data.keys())==sorted(['fromMote', 'toMote'])
-        
+
         fromMote = int(data['fromMote'])
         toMote   = int(data['toMote'])
-        
+
         self.engine.propagation.deleteConnection(fromMote,toMote)
-    
+
     def _topologyRouteRetrieve(self):
-        
+
         data = bottle.request.query
-        
+
         assert data.keys()==['destination']
-        
+
         detination_eui = [0x14,0x15,0x92,0xcc,0x00,0x00,0x00,int(data['destination'])]
-        
+
         route = self._dispatchAndGetResult(
-            signal       = 'getSourceRoute', 
+            signal       = 'getSourceRoute',
             data         = detination_eui,
         )
-        
+
         route = [r[-1] for r in route]
-        
+
         data = {
             'route'          : route,
         }
-        
+
         return data
-    
+
+    #@view('download.tmpl')
+    def _topologyDownload(self):
+        '''
+        Retrieve the topology data, in JSON format, and download it.
+        '''
+        data = self._topologyData()
+        now = datetime.datetime.now()
+        DAGrootList=[]
+
+        for ms in self.app.moteStates:
+            if(ms.getStateElem(moteState.moteState.ST_IDMANAGER).isDAGroot):
+                DAGrootList.append(ms.getStateElem(moteState.moteState.ST_IDMANAGER).get16bAddr()[1])
+
+        data['DAGrootList']=DAGrootList
```

Comment:
could you create an issue to replace the local `isDAGroot` variable by walking through the moteState?